clippy_lints/src/types.rs

```diff
@@ -211,6 +211,15 @@ fn check_ty(cx: &LateContext, ast_ty: &hir::Ty, is_local: bool) {
                         let PathParameters::AngleBracketedParameters(ref ab_data) = bx.parameters,
                         let [ref inner] = *ab_data.types
                     ], {
+                        if_let_chain! {[
+                            let ty::TyDynamic(binder, _) = cx.tables.node_id_to_type(ty.id).boxed_ty().sty,
```

Comment:
hmm... it seems rustc doesn't store the types anymore. `node_id_to_type` seems to panic.

We could try to do the same as above:

1. destructure TyPath (if it's not a path, it can't be `Any` anyway)
2. get the `def_id` via the `qpath_def` method and the `opt_def_id` function
3. call the `match_def_path` as you are doing it below.